modal/sandbox.py

```diff
@@ -388,6 +389,38 @@ def returncode(self) -> Optional[int]:
         else:
             return self._result.exitcode
 
+    @staticmethod
+    async def list(
+        *, app_id: Optional[str] = None, client: Optional[_Client] = None
+    ) -> AsyncGenerator["_Sandbox", None]:
+        """List all sandboxes for the current environment or app ID (if specified). Returns an iterator over `Sandbox` objects."""
+        before_timestamp = time.time()
```

Comment:
Is there any risk of weird behavior if there is misalignment between the clock in the local process and on the server? Since you start with the local time but then use the server-sourced time for subsequent batches. I'm not sure I have a super specific failure mode in mind, so maybe not a problem. But I wonder if you risk ending up missing some or including some multiple times or something like that.

Reply:
Since this is only used for the initial fetch and all times are treated as the time recorded by postgres, we would never get an inconsistent result. If the clock of the client is really far behind, we just may not get all of the active sandboxes. This issue would still exist even if we let the server pick the initial time because its clock could be behind.

Reply:
Hm but if we don't send a `before_timestamp`, the server doesn't use it in the query, so just setting it to `None` initially seems better I think?

Reply:
Fair point, done!